### PR TITLE
Adding MLE Coding test of cats and dogs

### PR DESCRIPTION
**Pull request by suneeta-mall**
https://xkcd.com/1558/

Chose this one ^^ curtsey @andychisholm !

### CODE REVIEW

File: cats_and_dogs.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+![https://xkcd.com/1558/](https://imgs.xkcd.com/comics/vet.png)
+
+# Find me pets
```
Comment (andychisholm) [inline cats_and_dogs.md]: Should this title perhaps be "Cats and dogs" to match the problem name?

```suggestion
# Cats and dogs
```

Comment (andychisholm) [reply cats_and_dogs.md]: For style points, suggest moving the heading above the image to (consistent with the formatting for the other problems)

File: cats_and_dogs.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+While we are not expecting you to deliver a production-ready model or solution, we would like to see the approach and thought process akin to a production solution building. To be more explicit:
+1. We don't want you to build the best-performing model to break some hypothetical leaderboard. We just want to see your approach to breaking down this problem, and how you choose the configuration (data, parameters, hypermeters, layout and algorithms) of the model. If your code runs and can do 1 epoch at least then it's acceptable. If you love cats/dogs and want to spend more time then you are welcome to show your passion there. Having said that, we respect your time and a workable solution is acceptable. 
+2. We would like to see your software engineering best practices but we certainly don't expect you to deliver zero bug solutions. How you prioritise issues and opportunities in your solution will be of keen interest to us. 
+Please do not hesitate to call out, if you are aware of things you would like to do if there were infinite hours in a day.
+3. Please provide your solution such that we can run it in our environments/machines independently. 
+
+
+## Confidentiality
```
Comment (andychisholm) [inline cats_and_dogs.md]: This section may not be up to date? Just noting the `hai-tech-tasks` is a public repo.

We also share some advice with candidates how to share their repo privately with the relevant hiring team, so might be able to omit it from here.

File: cats_and_dogs.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+1. What animals are present in the image? A cat or a dog or both
+2. Identify the breeds of the animals present in the image. 
+3. Provide a binary mask presenting which pixel features a pet (cat or dog) and which does not.
+
+We won't prescribe the API specification or expected response structure. We would like to see your approach to designing this API. 
+
+### Bonus 
+
+There will be some brownie points if you can write a short blurb on how you would scale your solution to:
```
Comment (andychisholm) [inline cats_and_dogs.md]:
```suggestion
There will be some brownie points if you can write a short blurb in your README.md on how you would scale your solution to:
```

**Review by andychisholm — APPROVED** [review]
LGTM!

File: cats_and_dogs.md (hunk 2 above)
Comment (suneeta-mall) [reply cats_and_dogs.md]: Ah Copy paste always gets one! ta

**Pull request merged**